toolchain/driver/llvm_subcommand.cpp

```diff
@@ -0,0 +1,65 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/driver/llvm_subcommand.h"
+
+#include "common/command_line.h"
+#include "llvm/TargetParser/Host.h"
+#include "llvm/TargetParser/Triple.h"
+#include "toolchain/base/llvm_tools.h"
+#include "toolchain/driver/llvm_runner.h"
+
+namespace Carbon {
+
+static constexpr CommandLine::CommandInfo SubcommandInfo = {
+    .name = "llvm",
+    .help = R"""(
+Runs LLVM's command line tools with the provided arguments.
+
+This subcommand provides access to a collection of LLVM's command line tools via
```

Comment:
This is probably future work, but it would probably be helpful to have some way of getting a list of the tools that can be invoked this way.

Reply:
The command line library actualy does this:

```
Runs LLVM's command line tools with the provided arguments.

This subcommand provides access to a collection of LLVM's command line tools via further subcommands. For each of these tools, their command line can be provided using positional arguments.

Subcommand `llvm` usage:
  carbon [OPTIONS] llvm ar [<ARG>...]
  carbon [OPTIONS] llvm cgdata [<ARG>...]
  carbon [OPTIONS] llvm cxxfilt [<ARG>...]
  carbon [OPTIONS] llvm debuginfod-find [<ARG>...]
...
```

It's not great though, because it just ends up with a generic tool help string for each and ever tool. So there is definitely some future work to make this more user friendly. I've added a TODO about that and the weird layering here.